lib/mpool/src/mdc.c

```diff
@@ -378,7 +378,7 @@ mpool_mdc_read(struct mpool_mdc *mdc, void *data, size_t len, size_t *rdlen)
     merr_t err;
     bool   verify = false;
 
-    if (!mdc || !data)
+    if (!mdc)
```

Comment:
Let `mdc_file_read` validate `len` and `data`.

Reply:
1. This is mpool, not HSE (public API) layer.
2. This function doesn't even use that parameter except to pass it down.
3. The validation has changed from simply `!data` to `(len && !data)` which is validated in the lower layer.

Reply:
Right, but what I am saying is that in HSE we validate all parameters regardless of use in kvdb_interface.c. So consistency would be validating everything in mpool.c. But I could also see the argument of pushing parameter validation to where it is used too

Reply:
Mpool has a lot of parameter validation because we were going to support apps written directly to mpool's API.  The file you mention, `kvdb_interface.c`, sits right below our public API which is why it validates all parameters.